### PR TITLE
Adds module selection to R commander

### LINKED ISSUE #1912: [Bug]: Allow for choosing jasp Module in r-commander
### JASP Version

0.16.4

### Commit ID

_No response_

### JASP Module

Other Module

### What analysis are you seeing the problem on?

_No response_

### What OS are you seeing the problem on?

Windows 10, macOS Intel

### Bug Description

Using R beta in 0.16.4 does not allow access to R packages that are used in JASP where previous versions (including 0.16.3) do. Commands such as library(BayesFactor) and other calls to packages do not work. This happens on both Mac (intel) and Windows versions.
JASP issue with R beta

### Expected Behaviour

The packages selected should load and be used for additional analysis using R code as they have with previous versions.

### Steps to Reproduce

1. Open R (beta)
2. Type the following code: library(BayesFactor)
3. Error: there is no package called ‘BayesFactor’
...

### Log (if any)

JASP 2022-11-22 00_11_11 Desktop.log

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [ ] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

### PR DESCRIPTION
**Pull request by JorisGoosen**
Also some feedback on what the engine is doing while the user waits. 
Fixes https://github.com/jasp-stats/jasp-issues/issues/1912

### CODE REVIEW

File: Desktop/components/JASP/Widgets/RCommanderWindow.qml [hunk 1]
```diff
@@ -259,7 +259,7 @@ Window
 				text:			qsTr("Run Code")
 				onClicked:		runCode();
 				width:			clearOutput.width
-				height:			(codeRect.height - 2 * jaspTheme.generalAnchorMargin) / 3
+				height:			(codeRect.height - 2 * jaspTheme.generalAnchorMargin) / 4
```
Comment (boutinb) [inline Desktop/components/JASP/Widgets/RCommanderWindow.qml]: It must be:

`(codeRect.height - 3 * jaspTheme.generalAnchorMargin) / 4`

File: Desktop/components/JASP/Widgets/RCommanderWindow.qml [hunk 2]
```diff
@@ -326,6 +326,24 @@ Window
 					margins:	jaspTheme.generalAnchorMargin
 				}
 			}
+
+			JC.DropDown
+			{
+				id:							selectModule
+				values:		 				dynamicModules.loadedModulesTitles
+				//startValue:				 	"Module selection"
```
Comment (boutinb) [inline Desktop/components/JASP/Widgets/RCommanderWindow.qml:333]: If you want a place holder use the properties:
```
placeholderText:			qsTr("Module selection")
addEmptyValue:			true
```

File: Desktop/components/JASP/Widgets/RCommanderWindow.qml [hunk 3]
```diff
@@ -326,6 +326,24 @@ Window
...
+				onValueChanged:				rCmd.loadModule(dynamicModules.loadedModules[currentIndex])
+				height:						runButton.height
```
Comment (boutinb) [inline Desktop/components/JASP/Widgets/RCommanderWindow.qml]: The height does not change the height of the inner control (maybe not what is expected...). To change the height of the inner control:
`control.height:			runButton.height`

**Review by boutinb — CHANGES_REQUESTED** [review]
I have also remarked that if the output is quite long then you can scroll to the top of the output. This is not due to this PR, but maybe a good occasion to fix this.

File: Desktop/components/JASP/Widgets/RCommanderWindow.qml (hunk 1 above)
Comment (JorisGoosen) [reply Desktop/components/JASP/Widgets/RCommanderWindow.qml]: can you make it a suggestion 
image

Comment (JorisGoosen) [reply Desktop/components/JASP/Widgets/RCommanderWindow.qml]:
```suggestion
				height:			(codeRect.height - 3 * jaspTheme.generalAnchorMargin) / 4
```

File: Desktop/components/JASP/Widgets/RCommanderWindow.qml (hunk 2 above)
Comment (JorisGoosen) [reply Desktop/components/JASP/Widgets/RCommanderWindow.qml:333]: Yeah I wanted to do that but now it just selects the first module anyway which is also fine by me.

File: Desktop/components/JASP/Widgets/RCommanderWindow.qml (hunk 3 above)
Comment (JorisGoosen) [reply Desktop/components/JASP/Widgets/RCommanderWindow.qml]: ah yes ok
```suggestion
				control.height:						runButton.height
```

**Pull request merged**